FBSDKCoreKit/FBSDKCoreKit/FBSDKApplicationDelegate.m

```diff
@@ -420,6 +420,11 @@ - (void)_logIfAppLinkEvent:(NSURL *)url
     return;
   }
 
+  NSString * targetURLString = applinkData[@"target_url"];
+  if (targetURLString == nil || ![targetURLString isKindOfClass: [NSString class]]) {
+    return;
+  }
+
   NSURL *targetURL = [NSURL URLWithString:applinkData[@"target_url"]];
```

Comment:
Thanks for the comment. I definitely see the reason behind your suggestion. We can avoid early exiting this method by being a bit cleverer. :) 

With the snippet above, there's a possible situation that `targetURL` is initialized with an `nil`. 
You can write `[NSURL URLWithString: nil];` and it compiles, but Xcode will give you this warning `Null passed to a callee that requires a non-null argument`. Xcode is telling us we *should* not pass an `nil` value into that method and I do like to keep the warnings in a codebase as few as possible and to call methods properly.

I think I might have an solution to avoid that situation (initializing `NSURL` with `nil`).

What do you think of the following?

```objective-c
  NSURL *targetURL;
  NSString * targetURLString = applinkData[@"target_url"];

  if (targetURLString == nil || ![targetURLString isKindOfClass: [NSString class]]) {
     targetURL = nil;
  } else {
    targetURL = [NSURL URLWithString:applinkData[@"target_url"]];
  }
```

This way, we do not early exit this method even when `targetURLString` is nil or is not of type `NSString` and we do not crash the SDK plus the apps.